### PR TITLE
Fixed inaccurate account balances in validator

### PR DESCRIPTION
**Pull request by ethDreamer**
When staking with multiple public keys, the validator wasn't tracking
each account's balance individually.

TODO(you): choose "part of" or "resolves" and the associated github issue #.

[Resolves] #2772 

---

# Description

**Write why you are making the changes in this pull request**

**Write a summary of the changes you are making**

**Link anything that would be helpful or relevant to the reviewers**

### CODE REVIEW

File: validator/client/validator.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type validator struct {
 	attesterClient       pb.AttesterServiceClient
 	keys                 map[string]*keystore.Key
 	pubkeys              [][]byte
-	prevBalance          uint64
+	prevBalance          []uint64
```
Comment (terencechain) [inline validator/client/validator.go]:
```suggestion
	prevBalances          []uint64
```

**Comment by cyberbono3** [comment]
Travis says "package github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1_gateway: cannot find package"

File: validator/client/validator_metrics.go [hunk 2]
```diff
@@ -24,13 +24,14 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 		return nil
 	}
 
-	epoch := slot / params.BeaconConfig().SlotsPerEpoch
-	if epoch == params.BeaconConfig().GenesisEpoch {
-		v.prevBalance = params.BeaconConfig().MaxDepositAmount
+	if (len(v.prevBalance) == 0) {
```
Comment (prestonvanloon) [inline validator/client/validator_metrics.go]:
```suggestion
	if (len(v.prevBalances) == 0) {
```

File: validator/client/validator_metrics.go [hunk 3]
```diff
@@ -24,13 +24,14 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
...
+		for i := 0; i < len(v.pubkeys); i++ {
+			v.prevBalance = append(v.prevBalance, params.BeaconConfig().MaxDepositAmount);
```
Comment (prestonvanloon) [inline validator/client/validator_metrics.go]:
```suggestion
			v.prevBalances = append(v.prevBalances, params.BeaconConfig().MaxDepositAmount);
```

File: validator/client/validator_metrics.go [hunk 4]
```diff
@@ -61,8 +62,8 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 		}
 		newBalance := float64(resp.Balance) / float64(params.BeaconConfig().GweiPerEth)
 
-		if v.prevBalance > 0 {
-			prevBalance := float64(v.prevBalance) / float64(params.BeaconConfig().GweiPerEth)
+		if v.prevBalance[i] > 0 {
```
Comment (prestonvanloon) [inline validator/client/validator_metrics.go]:
```suggestion
		if v.prevBalances[i] > 0 {
```

File: validator/client/validator_metrics.go [hunk 5]
```diff
@@ -61,8 +62,8 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
...
+			prevBalance := float64(v.prevBalance[i]) / float64(params.BeaconConfig().GweiPerEth)
```
Comment (prestonvanloon) [inline validator/client/validator_metrics.go]:
```suggestion
			prevBalance := float64(v.prevBalances[i]) / float64(params.BeaconConfig().GweiPerEth)
```

File: validator/client/validator_metrics.go [hunk 6]
```diff
@@ -71,10 +72,14 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 				"percentChange": fmt.Sprintf("%.5f%%", percentNet*100),
 				"pubKey":        tpk,
 			}).Info("Net gains/losses in eth")
+			// Remove the line below this comment if you want the prevBalance to be the
+			// initial balance so that the log actually shows the NET gains/losses since
+			// the accounts started staking. Otherwise it will show gains/losses since the
+			// previous block.
+			v.prevBalance[i] = resp.Balance
```
Comment (prestonvanloon) [inline validator/client/validator_metrics.go]:
```suggestion
			v.prevBalances[i] = resp.Balance
```

**Comment by prestonvanloon** [comment]
@ethDreamer We don't have permission to push to your branch. If you can accept the suggestions in comments then we can merge this.

**Pull request closed without merging**